enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,327 @@
+---
+title: azure_user_defined_tags
+authors:
+  - "@bhb"
+reviewers:
+  - "@patrickdillon" ## reviewer for installer component
+  - "@JoelSpeed" ## reviewer for api and machine-api-provider-azure components
+  - "@dmage" ## reviewer for cluster-image-registry-operator component
+  - "@Miciah" ## reviewer for cluster-ingress-operator component
+  - "@akhil-rane" ## reviewer for cloud-credential-operator component
+  - "@trozet" ## reviewer for cloud-network-config-controller component
+approvers:
+  - "@jerpeter1" ## approver for CFE
+api-approvers:
+  - "@JoelSpeed" ## approver for api component
+creation-date: 2022-07-12
+last-updated: 2022-07-12
+tracking-link:
+  - https://issues.redhat.com/browse/OCPPLAN-8155
+  - https://issues.redhat.com/browse/CORS-2249
+see-also:
+  - "enhancements/api-review/custom-tags-aws.md"
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# Apply user defined tags to all Azure resources created by OpenShift
+
+## Summary
+
+This enhancement describes the proposal to allow an administrator of Openshift to 
+have the ability to apply user defined tags to those resources created by Openshift 
+in Azure.
+
+## Motivation
+
+Motivations include but are not limited to:
+
+- Allow admin, compliance, and security teams to keep track of assets and objects 
+  created by OpenShift in Azure.
+
+### User Stories
+
+- As an openshift administrator, I want to have tags added to all resources created 
+  in Azure by Openshift, so that I can restrict access granted to an OpenShift specific account.
+
+### Goals
+
+- The administrator or service (in the case of Managed OpenShift) installing OpenShift 
+  can configure allowed number of user-defined tags in the Openshift installer generated
+  install config, which is referred and applied by the installer and the in-cluster operators
+  on the the Azure resources during cluster creation.
+- Tags must be applied at creation time, in an atomic operation. It isn't acceptable 
+  to create an object and to apply tags post cluster creation.
+
+### Non-Goals
+
+- Management(update/delete) of resource tags post creation of cluster is out of scope.
+
+## Proposal
+
+A tag of the form `kubernetes.io_cluster.<cluster_id>:owned` will be added to every
```

Comment:
This tag has changed since the initial revision of the enhancement, can you quickly explain why? Is the tag format requirement different in Azure than AWS?

Reply:
A mistake from me, azure tag names cannot have `/` character and at present tag added to Azure ResourceGroup is of the `kubernetes.io_cluster.<cluster_id>:owned` form.

Reply:
I'm confused.  Does the installer already add the `kubernetes.io_cluster.<cluster_id>:owned` tag?  It looks like it does: https://github.com/openshift/installer/blob/face870099c9408541367651b2d1f45ab7a86a01/pkg/asset/cluster/azure/azure.go#L55

If I understand correctly, it is already an existing, well known tag.  Are you trying to say that (1) the installer will continue to add the `kubernetes.io_cluster.<cluster_id>:owned` tag (i.e. the EP doesn't change this existing behavior), (2) now operators will also add the `kubernetes.io_cluster.<cluster_id>:owned` tag, and (3) the installer and operators will also add the tags from `userTags`?

Reply:
@bharath-b-rh, I'm still confused on this point.

Reply:
@bharath-b-rh - I think the questions in this conversation are still unresolved.

Reply:
I had missed updating here, installer currently adds just the ocp tag `kubernetes.io_cluster.<cluster_id>:owned` to only `ResourceGroup` resource. With this EP installer and the operators will add ocp and user-defined tags to all the resources created by OpenShift.